src/native/s3select/s3select_napi.cpp

```diff
@@ -0,0 +1,251 @@
+    Napi::Promise::Deferred deferred;
+    s3selectEngine::csv_object& csv_object;
+    s3selectEngine::json_object& json_object;
+    std::string select;
+    const char* buffer;
+    const unsigned int buffer_len;
+    const bool is_flush;
+    const InputFormat input_format;
+    const u_int8_t* headers_bytes;
+    const u_int32_t headers_len;
+
+    SelectWorker(Napi::Env env,
+        s3selectEngine::csv_object& csv_object,
+        s3selectEngine::json_object& json_object,
+        const char* buffer,
+        const unsigned int buffer_len,
+        const bool is_flush,
+        const InputFormat input_format,
+        const u_int8_t* headers_bytes,
+        const u_int32_t headers_len)
+        : AsyncWorker(env)
+        , csv_object(csv_object)
+        , json_object(json_object)
+        , buffer(buffer)
+        , buffer_len(buffer_len)
+        , is_flush(is_flush)
+        , input_format(input_format)
+        , deferred(env)
+        , headers_bytes(headers_bytes)
+        , headers_len(headers_len)
+    {}
+
+    void Execute() override
+    {
+        this->select.clear();
+
+        int rc;
+        // TODO - if we get total size on beginning, i think we can remove the is_flush if
+        if (csv == input_format) {
+            if (is_flush) {
+                rc = csv_object.run_s3select_on_stream(select, buffer, 0, 0);
+            } else {
+                rc = csv_object.run_s3select_on_stream(select, buffer, buffer_len, SIZE_MAX);
+            }
+        } else {
+            if (is_flush) {
+                rc = json_object.run_s3select_on_stream(select, buffer, 0, 0);
+            } else {
+                rc = json_object.run_s3select_on_stream(select, buffer, buffer_len, SIZE_MAX);
+            }
+        }
+        if (rc < 0) {
+            if (csv == input_format) {
+                SetError(csv_object.get_error_description());
+            } else {
+                // SetError(json_object.get_error_description()); //TODO - they added a getter, use after submodule is updated
+                SetError("failed to select from json");
+            }
+        }
+
+        /*std::cout << "select res = " << this->select << std::endl;
+        std::cout.flush();*/
+    }
+
+    void OnOK() override
+    {
+        Napi::Env env = Env();
+        // in case of empty select result (ie, no rows in current buffer matched sql condition), return null
+        if (0 == select.length()) {
+            deferred.Resolve(env.Null());
+            return;
+        }
+
+        Napi::Object result = Napi::Object::New(env);
+        Napi::Value select_buf = Napi::Buffer<char>::/*New*/ Copy(env, select.data(), select.length());
+        result.Set("select", select_buf);
+
+        u_int32_t prelude[3];
+        int32_t prelude_crc, message_crc, prelude_crc_BE;
+        prelude[0] = htonl(select.length() + headers_len + 16);
+        prelude[1] = htonl(headers_len);
+        prelude_crc = crc32_gzip_refl_base(0, (u_int8_t*)prelude, 8);
```

Comment:
use the standard type name `uint8_t` which is defined in cstdint.h - not sure where this `u_int8_t` comes from but I would use the standard anyway.